tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedValueMatcher.cs

```diff
@@ -0,0 +1,214 @@
+    {
+        private EntityMap _entityMap;
+
+        public UnifiedValueMatcher(EntityMap entityMap)
+        {
+            _entityMap = entityMap;
+        }
+
+        public void AssertValuesMatch(BsonValue expected, BsonValue actual)
+        {
+            AssertValuesMatch(expected, actual, true);
+        }
+
+        private void AssertValuesMatch(BsonValue expected, BsonValue actual, bool isRoot)
+        {
+            if (expected.IsBsonDocument &&
+                expected.AsBsonDocument.ElementCount == 1 &&
+                expected.AsBsonDocument.GetElement(0).Name.StartsWith("$$"))
+            {
+                var expectedDocument = expected.AsBsonDocument; // TODO: Recheck if non document allowed
+                switch (expectedDocument.GetElement(0).Name)
+                {
+                    case "$$unsetOrMatches":
+                        // TODO: recheck
+                        if (actual == null)
+                        {
+                            return;
+                        }
+                        expected = expectedDocument[0];
+                        break;
+                    case "$$type":
+                        AssertExpectedType(actual, expectedDocument[0]);
+                        return;
+                    case "$$matchesHexBytes":
+                        // TODO: recheck
+                        expected = expectedDocument[0];
+                        break;
+                    default:
+                        throw new NotSupportedException($"Special operator not supported: '{expectedDocument.GetElement(0).Name}'");
+                }
+            }
+
+            if (expected.IsBsonDocument)
+            {
+                actual.IsBsonDocument.Should().BeTrue($"Actual value must be a document, but is '{actual.BsonType}'");
+
+                var expectedDocument = expected.AsBsonDocument;
+                var actualDocument = actual.AsBsonDocument;
+
+                foreach (var expectedElement in expectedDocument)
+                {
+                    BsonValue expectedItem = expectedElement.Value;
+                    // TODO: consider using var t = expectedElement.Value is BsonDocument expectedDocument1;
+                    if (expectedElement.Value.IsBsonDocument &&
+                        expectedElement.Value.AsBsonDocument.ElementCount == 1 &&
+                        expectedElement.Value.AsBsonDocument.GetElement(0).Name.StartsWith("$$"))
+                    {
+                        var specialOperator = expectedElement.Value.AsBsonDocument;
+                        switch (specialOperator.GetElement(0).Name) // TODO: Recheck the switch order
+                        {
+                            case "$$exists":
+                                actualDocument.Contains(expectedElement.Name).Should().Be(specialOperator[0].AsBoolean); // TODO: Recheck this actually works
+                                return; // TODO: eh. return or continue?
```

Comment:
`continue` is correct.   In Java it's a `return` because this block of code is in a lambda function that executes once per element, so return in that context is like continue here, where you're using a loop

Reply:
Deeply nested `continue` is VERY hard to understand.

What loop exactly is being continued?